nautobot/ui/src/components/FiltersPanel.js

```diff
@@ -128,6 +160,393 @@ export const FiltersPanelContainer = forwardRef(
+                          []
+                    : [];
+            } catch (exception) {}
+
+            return [];
+        }, [objectType]);
+
+        const [lastUsedFilters, setLastUsedFilters] = useState(
+            getInitialLastUsedFilters
+        );
+
+        const [lookupField, setLookupField] = useState(null);
+        const [lookupType, setLookupType] = useState(null);
+        const [lookupValue, setLookupValue] = useState("");
+
+        const lookupFieldOptions = useMemo(
+            () =>
+                Object.entries(lookupFields).map(([key, value]) => ({
+                    label: value.label,
+                    value: key,
+                })),
+            [lookupFields]
+        );
+
+        const fieldLookupChoices = useMemo(
+            () =>
+                lookupField
+                    ? lookupFields[lookupField].lookup_types.map((filter) => ({
+                          value: filter.value,
+                          label: filter.label,
+                      }))
+                    : [],
+            [lookupField, lookupFields]
+        );
+
+        const onChangeLookupField = useCallback(
+            (newValue) => setLookupField(newValue.value),
+            [setLookupField]
+        );
+
+        const onChangeLookupType = useCallback(
+            (newValue) => setLookupType(newValue.value),
+            [setLookupType]
+        );
+
+        const onChangeLookupValue = useCallback(
+            (event) => setLookupValue(event.currentTarget.value),
+            [setLookupValue]
+        );
+
+        const onSubmitFilter = useCallback(
+            (event) => {
+                event.preventDefault();
+
+                if (
+                    lookupField &&
+                    lookupType &&
+                    lookupValue &&
+                    !isFilterAlreadyApplied(lookupType, lookupValue)
+                ) {
+                    addActiveFilter({ label: lookupType, value: lookupValue });
+
+                    setLastUsedFilters((currentLastUsedFilters) =>
+                        currentLastUsedFilters.every(
+                            ({ label, value }) =>
+                                label !== lookupType || value !== lookupValue
+                        )
+                            ? [
+                                  { label: lookupType, value: lookupValue },
+                                  ...currentLastUsedFilters,
+                              ]
+                            : currentLastUsedFilters
+                    );
+                }
+            },
+            [
+                addActiveFilter,
+                isFilterAlreadyApplied,
+                lookupField,
+                lookupType,
+                lookupValue,
+                setLastUsedFilters,
+            ]
+        );
+
+        useEffect(() => {
+            setLookupField(null);
+        }, [lookupFields, setLookupField]);
+
+        useEffect(() => {
+            setLookupType(null);
```

Comment:
To automatically populate the `Lookup Type` if there's only one option:

```suggestion
            fieldLookupChoices.length === 1 ? setLookupType(fieldLookupChoices[0].value) : setLookupType(null);
```